### PR TITLE
pkg/machine/wsl: wrap command errors

### PR DESCRIPTION
**Pull request by Luap99**
First of some commands ignored cmd.Wait() error which means it was impossible to notice any command errors. And other just returned the wait error as it which is when a command fails just `exit status <code>` which is not helpful at all.

This commit should add proper error wrapping with stderr to get useful strings back hopefully.

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by Luap99** [comment]
```
  C> podman.exe machine rm --force m1-742391f9cdd7
  Error: exit status 124
```

Looks like I didn't catch the place where the test is failing, I have to keep looking...

**Comment by n1hility** [comment]
LGTM

File: pkg/machine/wsl/machine.go [hunk 1]
```diff
@@ -571,7 +578,10 @@ func pipeCmdPassThrough(name string, input string, arg ...string) error {
 	cmd.Stdin = strings.NewReader(input)
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
-	return cmd.Run()
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("command %s %v failed: %w", name, arg, err)
+	}
+	return nil
```
Comment (cevich) [inline pkg/machine/wsl/machine.go:584]: I realize you want to get CI un-stuck for everyone.  I counted three seemingly identical repetitions of this sequence in this file.  Would you consider a followup PR to wrap these up in a common function somewhere?

Comment (baude) [reply pkg/machine/wsl/machine.go:584]: i would agree that having context as part of the error message would be beneficial.

Comment (Luap99) [reply pkg/machine/wsl/machine.go:584]: putting this into a function just to deduplicate the error text? Or deduplicate much more, i.e. including the cmd.Run() part?

There is much that can be improved here but I honestly don't have the time for that currently and given I have no windows setup I rather not fiddle to much with wsl code as I cannot test locally.

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Comment by baude** [comment]
given this passed, im going to merge.  we can clean up error messages in subsequent submissions.   thanks @Luap99

**Pull request merged**